calender/calender.go

```diff
@@ -0,0 +1,395 @@
+package calender
+
+// TODO:
+// FA470: Appointment erst ab einem gegebenen Datum anezeigen lassen -> submission.visibleFrom
+//	"Checkbox, ob der Kursteilnehmer die Abgabe erst ab einem bestimmten Zeitpunkt sieht"
+//	"Wenn vorherige Checkbox checked ist: Zeitfeld und Uhrzeitfeld für Sichtbarkeitsdatum für Kursteilnehmer"
+// FA330: Gelangen zu Abgabe über den Kalender:
+//	"Endnutzer sieht auf seinem Dashboard und über den Navigationspunkt „Stundenplan“ seinen Kalender"
+//	"Durch Drücken auf einer der Deadlines für seine Abgaben,
+//		öffnet sich für den Endnutzer ein Formular zu dieser Abgabe mit dazugehörigen Informationen (siehe andere F.A.)"
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/volatiletech/null/v8"
+	"github.com/volatiletech/sqlboiler/v4/boil"
+	"learningbay24.de/backend/models"
+)
+
+/*Returns all appointments the user with the user-ID has*/
+func GetAllAppointments(db *sql.DB, userId int) ([]*models.Appointment, error) {
+	tx, err := db.BeginTx(context.Background(), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	user, err := models.FindUser(context.Background(), tx, userId)
+	if err != nil {
+		if e := tx.Rollback(); e != nil {
+			return nil, fmt.Errorf("fatal: unable to rollback transaction on error: %s; %s", err, e)
+		}
+		return nil, err
+	}
+	courseQuery := user.UserHasCourses()
+	courseSlice, err := courseQuery.All(context.Background(), tx)
+	if err != nil {
+		if e := tx.Rollback(); e != nil {
+			return nil, fmt.Errorf("fatal: unable to rollback transaction on error: %s; %s", e, e)
+		}
+		return nil, err
+	}
+	var allAppointments []*models.Appointment
+
+	// Collect all appointments from all courses of the user
+	for i := 0; i < len(courseSlice); i++ {
```

Comment:
Eine for-each Schleife macht klarer, was hier passiert. `range` gibt `(int, obj)`, also Index und das gerade iterierte Objekt zurück.
Gerne überall abändern, beugt auch Fehlern vor.

```go
for _, course := range courseSlice {
  /* ... */
}
```